sdk/formrecognizer/Azure.AI.FormRecognizer/src/CustomFormModel.cs

```diff
@@ -112,41 +136,88 @@ private static IReadOnlyList<CustomFormSubmodel> ConvertFromUnlabeled(KeysResult
+                    CalculateFieldMap(trainResult),
+                    trainResult.ModelId));
+            }
+
+            return submodels;
+        }
+
+        private static Dictionary<string, CustomFormModelField> CalculateFieldMap(TrainResult trainResult)
         {
             var fieldMap = new Dictionary<string, CustomFormModelField>();
 
-            if (model.TrainResult.Fields != null)
+            if (trainResult.Fields != null)
             {
-                foreach (var formFieldsReport in model.TrainResult.Fields)
+                foreach (var formFieldsReport in trainResult.Fields)
                 {
                     fieldMap.Add(formFieldsReport.Name, new CustomFormModelField(formFieldsReport.Name, null, formFieldsReport.Accuracy));
                 }
             }
 
-            return new List<CustomFormSubmodel> {
-                new CustomFormSubmodel(
-                    $"form-{model.ModelInfo.ModelId}",
-                    model.TrainResult.AverageModelAccuracy,
-                    fieldMap)};
+            return fieldMap;
+        }
+
+        private static IReadOnlyList<TrainingDocumentInfo> ConvertToTrainingDocuments(Model model)
+        {
+            // TrainResult can be null if model is not ready yet.
+            if (!model.ModelInfo.Properties.IsComposedModel && model.TrainResult != null)
+            {
+                return ConvertToTrainingDocumentsInfo(model.TrainResult);
+            }
+
+            if (model.ModelInfo.Properties.IsComposedModel && model.ComposedTrainResults != null)
+            {
+                var trainingDocs = new List<TrainingDocumentInfo>();
+
+                foreach (var trainResult in model.ComposedTrainResults)
+                {
+                    trainingDocs.AddRange(ConvertToTrainingDocumentsInfo(trainResult));
+                }
+
+                return trainingDocs;
+            }
+
+            return new List<TrainingDocumentInfo>();
         }
 
-        private static IReadOnlyList<TrainingDocumentInfo> ConvertToTrainingDocuments(TrainResult trainResult)
+        private static IReadOnlyList<TrainingDocumentInfo> ConvertToTrainingDocumentsInfo(TrainResult trainResult)
         {
             var trainingDocs = new List<TrainingDocumentInfo>();
-            foreach (var docs in trainResult.TrainingDocuments)
+            foreach (var document in trainResult.TrainingDocuments)
             {
                 trainingDocs.Add(
                     new TrainingDocumentInfo(
-                        docs.Name,
-                        docs.PageCount,
-                        docs.Errors ?? new List<FormRecognizerError>(),
-                        docs.Status));
+                        document.Name,
+                        document.PageCount,
+                        document.Errors ?? new List<FormRecognizerError>(),
+                        document.Status,
+                        trainResult.ModelId));
```

Comment:
For a model that is not composed model, the ModelId will appear in CustomFromModel and in TrainingDocuments.
I was torn between leaving `TrainingDocumentInfo.ModelId` null or with the repeated `ModelId`. I thought leaving it was not a big deal instead of users trying to distinguish when the value is null and not.